packages/client/auth/src/CrossmintAuthClient.ts

```diff
@@ -0,0 +1,182 @@
+import type { UseSignInData } from "@farcaster/auth-kit";
+import {
+    type AuthMaterialWithUser,
+    CROSSMINT_API_VERSION,
+    CrossmintAuth,
+    type OAuthProvider,
+    REFRESH_TOKEN_PREFIX,
+    SESSION_PREFIX,
+} from "@crossmint/common-sdk-auth";
+import type { Crossmint } from "@crossmint/common-sdk-base";
+import { type CancellableTask, queueTask } from "@crossmint/client-sdk-base";
+import { deleteCookie, getCookie, getJWTExpiration, setCookie, TIME_BEFORE_EXPIRING_JWT_IN_SECONDS } from "./utils";
+
+export class CrossmintAuthClient extends CrossmintAuth {
+    private callbacks: CrossmintAuthClientCallbacks;
+    private refreshTask: CancellableTask | null = null;
+    private isRefreshing = false;
+
+    private constructor(crossmint: Crossmint, callbacks: CrossmintAuthClientCallbacks = {}) {
+        super(crossmint);
+        this.callbacks = callbacks;
+        this.handleRefreshToken();
+    }
+
+    public static from(crossmint: Crossmint, callbacks?: CrossmintAuthClientCallbacks): CrossmintAuthClient {
+        return new CrossmintAuthClient(crossmint, callbacks);
+    }
+
+    public getSession() {
+        this.handleRefreshToken();
+        return getCookie(SESSION_PREFIX);
+    }
+
+    public async getUser() {
+        const result = await this.apiClient.get(`api/${CROSSMINT_API_VERSION}/sdk/auth/user`, {
+            headers: {
+                "Content-Type": "application/json",
+            },
+        });
+
+        const user = await result.json();
+        return user;
+    }
+
+    public storeAuthMaterial(authMaterial: AuthMaterialWithUser) {
+        setCookie(SESSION_PREFIX, authMaterial.jwt);
+        setCookie(REFRESH_TOKEN_PREFIX, authMaterial.refreshToken.secret, authMaterial.refreshToken.expiresAt);
+    }
+
+    public logout() {
+        deleteCookie(REFRESH_TOKEN_PREFIX);
+        deleteCookie(SESSION_PREFIX);
+        this.callbacks.onLogout?.();
+    }
+
+    public async handleRefreshToken(refreshTokenSecret?: string): Promise<void> {
```

Comment:
this method is `async`, but its promise is usually ignored.
I don't know the details of how memory management works client side, but is it safe to let a promise go out of scope without any references? In other ~more serious~ languages, a Garbage Collector would consider that object orphaned and clean it up.

Reply:
copying from another comment:

> handleRefreshToken triggers a task to get the most recent auth material and set off a timer to re-check later on so it's more like a background process that we want to make sure it's being called on init.

That's why this doesn't return anything

Reply:
I understand it doesn't return anything, but the promise is still running in the background.
Given I'm not familiar with JS, I asked GPT and this is what I got:

**TL;DR: You can safely ignore the `Promise<void>` response, as JS is just.. JS.**

> No, JavaScript’s garbage collector will not stop a promise that is still pending or hasn’t resolved yet, even if it becomes unreachable.
> 
> In JavaScript, promises are designed to run to completion once they are created. This means that when a promise is created, the asynchronous operation it represents will proceed independently of whether there are references to the promise object. If the promise resolves (or rejects), it will do so regardless of whether any code is waiting for it or whether it has been stored in a variable.
> 
> Here’s what happens in two main cases:
> 
> 	1.	When a promise is unreachable but pending: JavaScript will not garbage collect the underlying asynchronous operation (such as a network request or a timeout) until it completes. So, the promise will still resolve or reject when that operation finishes.
> 	2.	When a promise is unreachable and resolved: Once a promise has resolved or rejected, and if there are no references to it, it can be garbage collected like any other object.
> 
> This approach ensures the reliability of asynchronous operations in JavaScript.

Reply:
Yeah, for further context, all async tasks enter the event loop. It has different stacks that as it loops around picks up a task from each stack. Promises I believe enter the microTask stack and remain there until resolved or rejected. Sorry, wasn't sure what you were referring to.

P.D. In case you want to learn more, this video explains it best: https://www.youtube.com/watch?v=cCOL7MC4Pl0